### ISSUE TITLE
How does this compare to scikit-learn BallTree?

### ISSUE THREAD

**Issue by blaylockbk** [tchlux/balltree#2]
Just a quick question, and this seemed like the best place to ask it...

How is this different than BallTree in scikit-learn? Is this faster?
https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.BallTree.html

**Comment by tchlux** [comment]
Running the `test.py` file on my machine gives these results (image). With 1 million points each having dimension 100, this code is nearly 10× faster to construct a tree and more than 50% faster to query exact neighbors.

I don't remember precisely how Vanderplas implemented the scikit ball tree, but I think the splitting metric was different. This code starts with a point on the convex hull, then does a binary split about the median based on 2-norm distance. This code also uses OpenMP to parallelize the tree construction process, where the scikit code is purely serial (last time I checked).

You can play with the size and dimension of the data and the number of nearest neighbors in the `test.py` file to see if it is faster on your hardware and for your application. Happy to help if you encounter issues! 

<img width="639" alt="image" src="https://user-images.githubusercontent.com/7769932/178599694-d1a67859-4927-4466-b031-f16676094f81.png">

**Comment by blaylockbk** [comment]
Thanks for the info!
